test/tests/functional/pbs_ralter.py

```diff
@@ -1516,3 +1550,254 @@ def test_failed_ralter(self):
+        """
+        offset = 20
+        duration = 20
+        shift = 15
+        rid, start, end = self.submit_and_confirm_reservation(offset,
+                                                              duration,
+                                                              standing=True)
+
+        self.alter_a_reservation(rid, start, end, shift=shift,
+                                 a_duration=30, check_log=False)
+
+        t_duration, t_start, t_end = self.get_resv_time_info(rid)
+        self.assertEqual(t_duration, 30)
+
+        # Wait for the reservation to start running.
+        self.check_resv_running(rid, offset - shift)
+
+        # Wait for the reservation occurrence to finish.
+        new_duration = t_end - t_start
+        self.check_occr_finish(rid, new_duration)
+
+        # Check that duration of the second occurrence is not altered.
+        self.check_standing_resv_second_occurrence(rid, start, end)
+
+    def test_standing_resv_duration_and_endtime(self):
+        """
+        This test case covers the below scenarios for a standing reservation.
+
+        1. Change duration and endtime of standing reservation
+        2. After the first occurrence of the reservation finishes, verify that
+           the start and end time of the second occurrence have not changed.
+
+        All the above operations are expected to be successful.
+        """
+        offset = 20
+        duration = 20
+        shift = 15
+        rid, start, end = self.submit_and_confirm_reservation(offset,
+                                                              duration,
+                                                              standing=True)
+
+        self.alter_a_reservation(rid, start, end, shift=shift, alter_e=True,
+                                 a_duration=30, check_log=False)
+
+        t_duration, t_start, t_end = self.get_resv_time_info(rid)
+        self.assertEqual(t_duration, 30)
+        self.assertEqual(t_end, end+15)
+        self.assertEqual(t_start, start+5)
+
+        # Wait for the reservation to start running.
+        self.check_resv_running(rid, offset - shift)
+
+        # Wait for the reservation occurrence to finish.
+        new_duration = t_end - t_start
+        self.check_occr_finish(rid, new_duration)
+
+        # Check that duration of the second occurrence is not altered.
+        self.check_standing_resv_second_occurrence(rid, start, end)
+
+    def test_standing_resv_duration_and_starttime(self):
+        """
+        This test case covers the below scenarios for a standing reservation.
+
+        1. Change duration and endtime of standing reservation
+        2. After the first occurrence of the reservation finishes, verify that
+           the start and end time of the second occurrence have not changed.
+
+        All the above operations are expected to be successful.
+        """
+        offset = 20
+        duration = 20
+        shift = 15
```

Comment:
Consider adding the new duration up here.  This way you can use it in the alter call plus the assert().  Also use these as much as possible in the below asserts (here and other tests).